code/addons/test/src/manager.tsx

```diff
@@ -197,30 +197,35 @@ addons.register(ADDON_ID, (api) => {
         );
       },
 
-      mapStatusUpdate: (state) =>
-        Object.fromEntries(
-          (state.details.testResults || []).flatMap((testResult) =>
-            testResult.results
-              .map(({ storyId, status, testRunId, ...rest }) => {
-                if (storyId) {
-                  const statusObject: API_StatusObject = {
-                    title: 'Component tests',
-                    status: statusMap[status],
-                    description:
-                      'failureMessages' in rest && rest.failureMessages?.length
-                        ? rest.failureMessages.join('\n')
-                        : '',
-                    data: {
-                      testRunId,
-                    },
-                    onClick: openAddonPanel,
-                  };
-                  return [storyId, statusObject];
-                }
-              })
-              .filter(Boolean)
-          )
-        ),
+      stateUpdater: (state, update) => {
+        if (update.details?.testResults) {
+          const statusUpdate = Object.fromEntries(
+            update.details.testResults.flatMap((testResult) =>
+              testResult.results
+                .map(({ storyId, status, testRunId, ...rest }) => {
+                  if (storyId) {
+                    const statusObject: API_StatusObject = {
+                      title: 'Component tests',
+                      status: statusMap[status],
+                      description:
+                        'failureMessages' in rest && rest.failureMessages?.length
+                          ? rest.failureMessages.join('\n')
+                          : '',
+                      data: {
+                        testRunId,
+                      },
+                      onClick: openAddonPanel,
+                    };
+                    return [storyId, statusObject];
+                  }
+                })
+                .filter(Boolean)
+            )
+          );
+          api.experimental_updateStatus(TEST_PROVIDER_ID, statusUpdate);
+        }
+        return { ...state, ...update };
+      },
```

Comment:
Why does this need to return anything at all?

We could/should just run this with the already updated state

Reply:
Yes, we could change the return value to `void | ...` so that when it doesn't return anything, it falls back to doing the default state update.